job_definitions/publish_draft_services.yml

```
@@ -23,8 +23,11 @@
           description: "The framework slug to publish services for, e.g. 'g-cloud-11'."
       - string:
           name: AWS_PROFILE
+          choices:
+            - production
+            - development
           default: "{% if environment == 'production' %}production{% else %}development{% endif %}"
```

I think choice parameters use a different syntax in Jenkins Job Builder ([docs](https://docs.openstack.org/infra/jenkins-job-builder/parameters.html#parameters.choice)).

The key needs to be `choice`, and I don't know if the value of the `default` key is respected, the docs just say that the first in the list is the default 🤔 .

```suggestion
      - choice:
          name: AWS_PROFILE
          choices:
            - production
            - development
          default: "{% if environment == 'production' %}production{% else %}development{% endif %}"
```

---

Ooo good spot. If there's no default wonder if I can do

```
- choice:
  name: AWS_PROFILE
  choices:
{% if environment == 'production' %}
     - production
     - development
{% else %}
     - development
     - production
{% endif %}
```

---

That should work!